README.md

```diff
@@ -1,164 +1,190 @@
+* [full resolution download option](https://github.com/publiclab/Leaflet.DistortableImage/pull/100) for large images, using WebGL acceleration
+
+[Download as zip](https://github.com/publiclab/Leaflet.DistortableImage/releases) or clone to get a copy of the repo.
+
+This plugin has basic functionality, and is in production as part of MapKnitter, but there are [plenty of outstanding issues to resolve](https://github.com/publiclab/Leaflet.DistortableImage/issues). Please consider helping out!
+
+The recommended Google satellite base layer can be integrated using this Leaflet plugin: https://gitlab.com/IvanSanchez/Leaflet.GridLayer.GoogleMutant
+
+Here's a screenshot:
+
+![screenshot](example.png)
+
+## Demo
+
+Check out this [simple demo](https://publiclab.github.io/Leaflet.DistortableImage/examples/index.html).
+
+And watch this GIF demo:
+
+![demo gif](https://raw.githubusercontent.com/publiclab/mapknitter/master/public/demo.gif)
+
+To test the code, open `index.html` in your browser and click and drag the markers on the edges of the image. The image will show perspectival distortions.
+
+## Usage
+
+```js
+// basic Leaflet map setup
+map = new L.map('map').setView([51.505, -0.09], 13);
+L.tileLayer('https://{s}.tiles.mapbox.com/v3/anishshah101.ipm9j6em/{z}/{x}/{y}.png', {
+  maxZoom: 18,
+  attribution: 'Map data &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors, ' +
+    '<a href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, ' +
+    'Imagery © <a href="http://mapbox.com">Mapbox</a>',
+  id: 'examples.map-i86knfo3'
+}).addTo(map);
+
+// create an image
+img = new L.DistortableImageOverlay(
+  'example.png', {
+    corners: [
+      new L.latLng(51.52,-0.10),
+      new L.latLng(51.52,-0.14),
+      new L.latLng(51.50,-0.10),
+      new L.latLng(51.50,-0.14)
+    ],
+    fullResolutionSrc: 'large.jpg', // optionally pass in a higher resolution image to use in full-res exporting
+    suppressToolbar: false // defaults to false, but you can turn off the toolbar interface and build your own
+  }
+).addTo(map);
+
+L.DomEvent.on(img._image, 'load', img.editing.enable, img.editing); // enable editing
+
+```
+
+## Full-resolution download
+
+We've added a GPU-accelerated means to generate a full resolution version of the distorted image; it requires two additional dependencies to enable; see how we've included them in the demo:
+
+
+```
+<script src="../node_modules/webgl-distort/dist/webgl-distort.js"></script>
+<script src="../node_modules/glfx-js/dist/glfx.js"></script>
+```
+## "Reveal" functionality
+
+The "reveal" tool extends the native `EditOverlayAction` Leaflet class behaviour and hides some user-specified tools, hence saving up space in the toolbar and giving more room to your mapview! Refer the comments below for a detailed explaination on what happens when the "reveal" tool is executed.
+
+```js
+function() {
+
+  /* specify an array of tools that will initially be

```

Comment:
There is not much context about using this - I think we should make this the underlying functionality behind a simple API that allows the user to define "extra" tools at initialization. Otherwise the user will have to dig around in source code. 
- Alternatively we should show an example of the array and the HTML and also clarify that the reveal tool is referenced as ToggleReveal, I think this could have been defined in the initial explanation paragraph

Reply:
Also just read the next section and this makes a lot more sense after that reading that one!! I think it is because you showed an initialization example. Let's add one here before moving on to custom toolbars! But you don't need to redefine the map.